server/filestore_test.go

```diff
@@ -6949,59 +6949,108 @@ func TestFileStoreRecoverWithRemovesAndNoIndexDB(t *testing.T) {
 ///////////////////////////////////////////////////////////////////////////
 
 func Benchmark_FileStoreSelectMsgBlock(b *testing.B) {
-	// We use small block size to create lots of blocks for this test.
-	fs, err := newFileStore(
-		FileStoreConfig{StoreDir: b.TempDir(), BlockSize: 128},
-		StreamConfig{Name: "zzz", Subjects: []string{"*"}, Storage: FileStorage})
-	require_NoError(b, err)
-	defer fs.Stop()
+	const Seed = 123456
+	cases := []struct {
+		blockSize uint64
+		msgSize   int
+		msgCount  int
+	}{
+		{128, 100, 1000},
+		{128, 1000, 1000}, // 10X msg size from base case
+		{128, 100, 10000}, // 10X msg count from base case
+	}
+
+	for _, params := range cases {
+
+		b.Run(
+			fmt.Sprintf("blockSize:%d,msgSize:%d,msgCount:%d", params.blockSize, params.msgSize, params.msgCount),
+			func(b *testing.B) {
+				fs, err := newFileStore(
+					FileStoreConfig{StoreDir: b.TempDir(), BlockSize: params.blockSize},
+					StreamConfig{Name: "zzz", Subjects: []string{"*"}, Storage: FileStorage})
+				require_NoError(b, err)
+				defer fs.Stop()
+
+				rng := rand.New(rand.NewSource(Seed))
+
+				// Add in a bunch of blocks.
+				msg := make([]byte, params.msgSize)
+				for i := 0; i < params.msgCount; i++ {
+					rng.Read(msg)
+					_, _, err := fs.StoreMsg("A", nil, msg)
+					require_NoError(b, err)
+				}
+				if fs.numMsgBlocks() < params.msgCount {
+					b.Fatalf("Expected at least 1000 blocks, got %d", fs.numMsgBlocks())
+				}
 
-	subj, msg := "A", bytes.Repeat([]byte("ABC"), 33) // ~100bytes
+				b.SetBytes(int64(params.msgSize))
 
-	// Add in a bunch of blocks.
-	for i := 0; i < 1000; i++ {
-		fs.StoreMsg(subj, nil, msg)
-	}
-	if fs.numMsgBlocks() < 1000 {
-		b.Fatalf("Expected at least 1000 blocks, got %d", fs.numMsgBlocks())
-	}
+				fs.mu.RLock()
+				defer fs.mu.RUnlock()
 
-	fs.mu.RLock()
-	defer fs.mu.RUnlock()
+				b.ResetTimer()
+				for i := 0; i < b.N; i++ {
+					_, mb := fs.selectMsgBlockWithIndex(1 + uint64(i%params.msgCount))
```

Comment:
n.b. this was always set to 1 in the original benchmark, now it rolls around, could also be random

Reply:
If the value has meaning for the test or benchmark I try to call it out.